### ISSUE TITLE
Use _PYTHON_SYSCONFIGDATA_NAME instead of _CONDA_PYTHON_SYSCONFIGDATA_NAME

### ISSUE THREAD

**Issue by isuruf** [conda-forge/python-feedstock#474]
Regarding patch https://github.com/conda-forge/python-feedstock/blob/master/recipe/patches/0010-Add-support-for-_CONDA_PYTHON_SYSCONFIGDATA_NAME-if-.patch

In our compilers package we set `_CONDA_PYTHON_SYSCONFIGDATA_NAME` and not `_PYTHON_SYSCONFIGDATA_NAME`.

Now that `setuptools` has vendored a copy of `distutils`, we need to apply this change there too. We haven't upgraded `setuptools`
to 50+ because we haven't patched it yet.

The issue with patching `setuptools` is that I came across an instance where a project was using conda for all the C libraries
and used conda's python to create a `venv` and installed other python packages using pip only. This resulted in a `setuptools` from `pip` which is `50+` and therefore unpatched.

I don't know the reason for `_CONDA_PYTHON_SYSCONFIGDATA_NAME` when `_PYTHON_SYSCONFIGDATA_NAME` is perfectly reasonable and then we don't need to patch `setuptools` because `_PYTHON_SYSCONFIGDATA_NAME` is honored by `distutils` in python and by the copy in `setuptools`.

Thoughts @conda-forge/core, @chenghlee ?

**Comment by chenghlee** [comment]
I don't know why this environment variable exists, either, but it's definitely something I'd like/need to dig into. I'm also wondering if the compiler activation scripts should be setting `_CONDA_PYTHON_SYSCONFIGDATA_NAME` at all.

This all came up when we noticed buggy behavior with `conda env` when it's set to a target triplet that doesn't match the target triplet used to build the `base` environment Python interpreter (conda/conda#10766).

**Comment by isuruf** [comment]
I think we can get away with removing `_CONDA_PYTHON_SYSCONFIGDATA_NAME` and `_PYTHON_SYSCONFIGDATA_NAME`.

The reason to have this was to have two different sysconfigdata.
1. For when conda compilers are used
2. For when system compilers are used.

There were lots of differences in the past, but right now the differences are as follows,
- 1 has CC=x86_64-conda-linux-gnu-gcc and 2 has CC=gcc
- 2 has `-B $PREFIX/compiler_compat -Wl,--sysroot=/`

I think for the first point, it's okay to have CC=gcc for both because when the conda compilers are used the env variable
CC is set and that takes precedence over what is in sysconfigdata.

For the second point, this was added so that even with the system compiler our linker is used because the conda libraries
are built with a newer binutils and you need a newer linker to handle new relocation patterns. Note that we have our linker in `$PREFIX/compiler_compat` and that's why the `-B` option is there and the sysroot option is because the linker needs to know
that the sysroot is the system sysroot. (`python` has a dependency on `ld_impl_linux-64` which gives `$PREFIX/bin/$HOST-ld` and there's a symlink from `$PREFIX/compiler_compat/ld -> $PREFIX/bin/$HOST-ld`)

I've added https://github.com/conda-forge/binutils-feedstock/pull/52 so that `-Wl,--sysroot=/` is not needed and the linker now falls back to system sysroot when the conda sysroot is not found.
So the only difference now is `-B $PREFIX/compiler_compat` which is okay to have even with conda compilers.

TL:DR; we can remove `_CONDA_PYTHON_SYSCONFIGDATA_NAME` and `_PYTHON_SYSCONFIGDATA_NAME` by combining the two sysroots.

**Comment by isuruf** [comment]
PR at #474

**Comment by isuruf** [comment]
@mbargull, where did you run into the error at https://github.com/conda-forge/admin-requests/pull/120?

**Comment by beckermr** [comment]
We had some work to do around this when we changed the sysroot name. I forget what exactly it was though. Do you recall @isuruf? Did we ever do that? Does this change make that obsolete?

**Comment by isuruf** [comment]
Yes, this makes it obsolete.

**Comment by mbargull** [comment]
> @mbargull, where did you run into the error at [conda-forge/admin-requests#120](https://github.com/conda-forge/admin-requests/pull/120)?

I can't recall, sorry (given the time frame, possibly during some bulk updates at Bioconda which don't have separate issues/PRs).
(But obviously this happened when a shared library/extension module got linked, so it shouldn't be too hard to find a test case for it.)

**Comment by isuruf** [comment]
Thanks @mbargull. Should be fixed with https://github.com/conda-forge/python-feedstock/pull/477

**Comment by mbargull** [comment]
> Thanks @mbargull. Should be fixed with #477

gh-477 might make sense either way (IIUC upstream only did it for Darwin because they only encountered `"gcc" != "clang"` therein at a larger scale but didn't consider other uncommon setups -- at least that was my impression).
But it I don't think it would fix the issue since the issue is rather that `setuptools >=50` shipped a copy of `distutils.sysconfig`, i.e., whatever we patch for `python`'s `distutils.sysconfig` we'd also have to patch at `setuptools._distutils.sysconfig`.

**Comment by isuruf** [comment]
We have removed _CONDA_PYTHON_SYSCONFIGDATA_NAME in 3.9 and 3.10.
See https://github.com/conda-forge/python-feedstock/issues/534 for 3.8
